CMakeLists.txt

```diff
@@ -68,59 +68,51 @@ target_include_directories(
 # Stub library for libtritonserver.so that stubs Triton Server API and
 # Triton Backend API
 #
-if(NOT WIN32)
-add_custom_command(
-  OUTPUT
-    tritonserver_stub.cc
-    COMMAND cp ${CMAKE_CURRENT_SOURCE_DIR}/include/triton/core/tritonserver.h tritonserver_stubpp
-    COMMAND sed -i '/\#/d' tritonserver_stubpp
-    COMMAND sed -i '/\\//d' tritonserver_stubpp
-    COMMAND sed -i '/Fn_t/d' tritonserver_stubpp
-    COMMAND sed -i -r 's/.*\(TRITONSERVER_[^\(]+\)\\\(.*/void \\1\(\)\{\}/g\;t\;d' tritonserver_stubpp
-    COMMAND cp ${CMAKE_CURRENT_SOURCE_DIR}/include/triton/core/tritonbackend.h tritonbackend_stubpp
-    COMMAND sed -i '/\#/d' tritonbackend_stubpp
-    COMMAND sed -i '/\\//d' tritonbackend_stubpp
-    COMMAND sed -i '/Fn_t/d' tritonbackend_stubpp
-    COMMAND sed -i -r 's/.*\(TRITONBACKEND_[^\(]+\)\\\(.*/void \\1\(\)\{\}/g\;t\;d' tritonbackend_stubpp
-    COMMAND sed -i '/Initialize\(/d' tritonbackend_stubpp
-    COMMAND sed -i '/Finalize\(/d' tritonbackend_stubpp
-    COMMAND sed -i '/ModelInstanceExecute\(/d' tritonbackend_stubpp
-    COMMAND echo 'extern \"C\" {' > tritonserver_stub.cc
-    COMMAND cat tritonserver_stubpp >> tritonserver_stub.cc
-    COMMAND cat tritonbackend_stubpp >> tritonserver_stub.cc
-    COMMAND echo '} /* extern \"C\" */' >> tritonserver_stub.cc
-)
-
 add_library(
   triton-core-serverstub SHARED
-  ${CMAKE_CURRENT_BINARY_DIR}/tritonserver_stub.cc
+  ${CMAKE_CURRENT_SOURCE_DIR}/tritonserver_stub.cc
 )
 
 add_library(
   TritonCore::triton-core-serverstub ALIAS triton-core-serverstub
 )
 
 target_compile_features(triton-core-serverstub PRIVATE cxx_std_11)
-target_compile_options(
-  triton-core-serverstub
-  PRIVATE
-    -Wall -Wextra -Wno-unused-parameter -Werror
-)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  message("Using MSVC as compiler, default target on Windows 10. "
+		  "If the target system is not Windows 10, please update _WIN32_WINNT "
+		  "to corresponding value.")
+  set(CMAKE_CXX_FLAGS "/W4 /D_WIN32_WINNT=0x0A00 /EHsc")
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
+
+  set_target_properties(
+    triton-core-serverstub
+    PROPERTIES
+      POSITION_INDEPENDENT_CODE ON
+      NO_SONAME ON
+      OUTPUT_NAME tritonserver
```

Comment:
Why are you giving the stub library a different name on windows?

Reply:
Windows doesn't have "soname", so the stub library need to be named the same as tritonserver so that the backend library doesn't try to look for tritonserver_stub.dll